src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -928,13 +964,11 @@ private void EnterParameter(ParameterSymbol parameter, TypeSymbolWithAnnotations
             {
                 if (EmptyStructTypeCache.IsTrackableStructType(parameterType.TypeSymbol))
                 {
-                    InheritNullableStateOfTrackableStruct(parameterType.TypeSymbol, slot, valueSlot: -1, isByRefTarget: parameter.RefKind != RefKind.None, slotWatermark: GetSlotWatermark());
+                    InheritNullableStateOfTrackableStruct(parameterType.TypeSymbol, slot, valueSlot: -1, isDefaultValue: false, isByRefTarget: parameter.RefKind != RefKind.None, slotWatermark: GetSlotWatermark());
```

Comment:
>isDefaultValue: false, [](start = 105, length = 22)

What about `void M(string s = default)`?
I think we report a warning already, but in terms of null-state the optional value should probably still count.

Reply:
Good catch, thanks.

---
In reply to: [246871918](https://github.com/dotnet/roslyn/pull/32304#discussion_r246871918) [](ancestors = 246871918)